.ci/setup.sh

```diff
@@ -11,15 +11,18 @@ ARCH=$(uname -m)
 
 
 if [[ $OS_NAME == "macos" ]]; then
+    # Check https://github.com/actions/runner-images/tree/main/images/macos for available
+    # versions of Xcode
+    macos_ver=$(sw_vers --productVersion)
+    if [[ "${macos_ver}" =~ 13. ]]; then
+        xcode_path="/Applications/Xcode_14.3.app/Contents/Developer"
+    else
+        xcode_path="/Applications/Xcode_15.0.app/Contents/Developer"
+    fi
```

Comment:
Chose the oldest included versions of XCode from here:

* https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md#xcode
* https://github.com/actions/runner-images/blob/main/images/macos/macos-14-Readme.md#xcode

And while doing this, proposing this more-direct approach for selecting the version... depending on the actual macOS version instead of things that indirectly select it from this project's CI configs, like `$AZURE` and `$COMPILER`.